javascript/funWithFunctions.md

```diff
@@ -0,0 +1,24 @@
+## Functional Expressions in D&D ##
```

Comment:
Based on the tone of the existing TILs, this seems a bit more "story-telly". The rest get straight to the point. Not sure if that's a guideline we want to follow with these, but it's something to consider. Thoughts @JoelQ?